### PR TITLE
Events widget public view

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #1184 :*
https://github.com/bcgov/met-public/issues/1184

-Add events widget to the public view

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/view/widgets/EventsWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                                    <MetBody>Address: {eventItem.location_address}</MetBody>
+                                </Grid>
+                            </When>
+                            <Grid item container justifyContent={{ xs: 'center', md: 'flex-start' }} xs={12}>
+                                <MetBody>Date: {formatDate(eventItem.start_date, 'MMMM DD, YYYY')}</MetBody>
+                            </Grid>
+                            <Grid container justifyContent={{ xs: 'center', md: 'flex-start' }} item xs={12}>
+                                <MetBody>
+                                    Time:{' '}
+                                    {`${formatDate(eventItem.start_date, 'h:mm a')} to ${formatDate(
```
Comment (saravanpa-aot) [inline met-web/src/components/engagement/view/widgets/EventsWidget.tsx:131]: any vetter way of handling formatDate ? without giving the format in the template..
like MMMM DD, YYYY will be in multiple places in the whole app

Comment (saravanpa-aot) [reply met-web/src/components/engagement/view/widgets/EventsWidget.tsx:131]: same with h:mm:a

File: met-web/src/components/engagement/view/widgets/EventsWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            </MetPaper>
+        );
+    }
+
+    if (events.length === 0) {
+        return null;
+    }
+
+    return (
+        <MetPaper elevation={1} sx={{ padding: '1em', minHeight: '12em' }}>
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/view/widgets/EventsWidget.tsx:66]: I would prefer breaking virtual and in person to separate components. and do a Case switch like in DocumentSwitch.tsx. This way they would be completely separate for maintenance. It's not a blocker though. I'll add approval and leave it to you.

**Review by jadmsaadaot — APPROVED** [review]

**Review by saravanpa-aot — APPROVED** [review]

File: met-web/src/components/engagement/view/widgets/EventsWidget.tsx (hunk 2 above)
Comment (djnunez-aot) [reply met-web/src/components/engagement/view/widgets/EventsWidget.tsx:66]: Ok I think that is a better idea for the future of this component. So I will work on that and get it merged. Thank you.

**Pull request merged**